### PR TITLE
Dread: Rest of Ferenia revision (E.M.M.I. Zone and Escue Area)

### LINKED ISSUE #3624: Dread: Wave EMMI room could be simplified
The Wave EMMI fight sequence involves many dangerous actions and could use some simplifications to minimize these.

```
>>> Player 0: From Burenia/Burenia Hub to Dairon/Pickup (Missile Tank), 2024 reachable nodes, 2014 safe nodes, 113 open pickup indices, 153 open events, pickups_left:
[E: Event - Chain Reaction Device] - 0.375
[E: Event - Rubble before Kraid] - 0.375
[E: Event - Ferenia Central Unit] - 0.375
[E: Event - Super Missile Rotatable] - 0.375

--> Collecting Ferenia/Central Unit/Event - Ferenia Central Unit

>>> Updating state of Player 1
-> New Events: Ferenia/Central Unit/Event - Ferenia Central Unit

===============================

>>> Player 0: From Ferenia/Central Unit/Event - Ferenia Central Unit, 6 reachable nodes, 5 safe nodes, 113 open pickup indices, 154 open events, pickups_left:
[E: Event - Lower Water] - 0.375

--> Collecting Ferenia/Central Unit Access/Event - Lower Water

>>> Updating state of Player 1
-> New Events: Ferenia/Central Unit Access/Event - Lower Water

===============================

>>> Player 0: From Ferenia/Central Unit Access/Event - Lower Water, 8 reachable nodes, 1 safe nodes, 113 open pickup indices, 155 open events, pickups_left:
[E: Event - Wide Beam Block 1] - 1.125

--> Collecting Ferenia/Purple EMMI Arena/Event - Wide Beam Block 1

>>> Updating state of Player 1
-> New Events: Ferenia/Purple EMMI Arena/Event - Wide Beam Block 1
-> New Events: Ferenia/Purple EMMI Arena/Event - Chase Blob 1 (db_reg_b2_011)
-> New Events: Ferenia/Purple EMMI Arena/Event - Chase Blob 2 (db_reg_b2_010)
-> New Events: Ferenia/Purple EMMI Arena/Event - Wide Beam Block 2
```

### PR DESCRIPTION
**Pull request by Hugoshido**
- Added some WBJ to the underwater section
- Changed most of the Spin jumps to trivial (Tested all of them myself)
- Changed all Walljumps (beginer) to trivial 
- Changed Chozo Twin arena and added fight template
- Added Chozo template for QR Event
- Redesigned the whole Purple EMMI Area (Deleted all notes and setup two main nodes for the whole room)

Fixes #3624

### CODE REVIEW

File: randovania/games/dread/json_data/Ferenia.txt [hunk 1]
```diff
@@ -150,7 +150,7 @@ Extra - asset_id: collision_camera_003
   * Extra - actor_name: dooremmy_002
   * Extra - actor_def: actordef:actors/props/dooremmy/charclasses/dooremmy.bmsad
   > Tunnel to Purple EMMI Introduction
-      Gravity Suit or Speed Booster
+      Gravity Suit or Speed Booster or Perform WBJ
```
Comment (XenoWars) [inline randovania/games/dread/json_data/Ferenia.txt:153]: Can add Space Jump at least. Will need to see if Spin Boost is enough

File: randovania/games/dread/json_data/Ferenia.txt [hunk 2]
```diff
@@ -189,9 +187,11 @@ Extra - asset_id: collision_camera_003
 > Center Platform; Heals? False
   * Layers: default
   > Dock to Teleport to Burenia (Cyan)
-      Grapple Beam or Gravity Suit or Use Spin Boost
+      Flash Shift or Grapple Beam or Gravity Suit or Spider Magnet or Lay Cross Comb
```
Comment (XenoWars) [inline randovania/games/dread/json_data/Ferenia.txt:190]: I would change Lay Cross Comb to Simple IBJ, and then add Grapple Movement to Grapple Beam. Also curious why you removed Use Spin Boost

File: randovania/games/dread/json_data/Ferenia.txt [hunk 3]
```diff
@@ -189,9 +187,11 @@ Extra - asset_id: collision_camera_003
...
+  > Door to EMMI Zone Exit Middle (Power)
+      Trivial
   > Door to EMMI Zone Exit Middle (Charge)
-      Grapple Beam or Gravity Suit or Use Spin Boost
+      Flash Shift or Grapple Beam or Gravity Suit or Spider Magnet or Lay Cross Comb
```
Comment (XenoWars) [inline randovania/games/dread/json_data/Ferenia.txt:194]: Same comments as above.

File: randovania/games/dread/json_data/Ferenia.txt [hunk 4]
```diff
@@ -1658,9 +1657,7 @@ Extra - asset_id: collision_camera_019
   > Event - Storm Missile Gate
       Activate Storm Missile Locks
   > Bomb Ledge
-      All of the following:
-          Screw Attack
-          Walljump (Beginner) or Use Spin Boost
+      Screw Attack
```
Comment (XenoWars) [inline randovania/games/dread/json_data/Ferenia.txt:1660]: I'm curious why there isn't a Screw Attack tile group node here. (nothing wrong here)

File: randovania/games/dread/json_data/Ferenia.txt [hunk 5]
```diff
@@ -2018,6 +2013,8 @@ Extra - asset_id: collision_camera_025
   * Extra - actor_def: actordef:actors/props/dooremmy/charclasses/dooremmy.bmsad
   > Door to Save Station Southeast
       Trivial
+  > Tile Group (MISSILE)
+      Trivial
```
Comment (XenoWars) [inline randovania/games/dread/json_data/Ferenia.txt:2017]: Personally, i feel this node can just be deleted. doesnt serve any purpose logically

File: randovania/games/dread/json_data/Ferenia.txt [hunk 6]
```diff
@@ -2294,7 +2293,9 @@ Extra - asset_id: collision_camera_028
   > Door to Cold Room (Energy Recharge Station)
       Trivial
   > Event - Wave Blob (db_reg_b2_005)
-      Wave Beam
+      Any of the following:
+          Wave Beam
+          Pseudo-Wave Beam (Beginner) and Shoot Beam
```
Comment (XenoWars) [inline randovania/games/dread/json_data/Ferenia.txt:2298]: Include Slide as a requirement

File: randovania/games/dread/json_data/Ferenia.txt [hunk 7]
```diff
@@ -2572,7 +2575,7 @@ Extra - asset_id: collision_camera_031
   > Door to EMMI Zone Exit East Access
       Can Slide
   > Tunnel to Purple EMMI Arena
-      Simple IBJ or Use Spin Boost
+      Flash Shift or Slide Jump (Beginner) or Simple IBJ or Use Spin Boost
```
Comment (XenoWars) [inline randovania/games/dread/json_data/Ferenia.txt:2578]: Include Slide for the Slide Jump. Also testing if you can just wall jump from a ledge grab would be worth something.

File: randovania/games/dread/json_data/Ferenia.txt [hunk 8]
```diff
@@ -2666,7 +2669,7 @@ Extra - asset_id: collision_camera_032
           Flash Shift or Spider Magnet or Space Jump
           All of the following:
               Gravity Suit
-              Simple IBJ or Use Spin Boost
+              Flash Shift or Simple IBJ or Use Spin Boost
```
Comment (XenoWars) [inline randovania/games/dread/json_data/Ferenia.txt:2672]: can take the flash back out of this

File: randovania/games/dread/json_data/Ferenia.txt [hunk 9]
```diff
@@ -2847,7 +2850,9 @@ Extra - asset_id: collision_camera_034
   * Extra - actor_layer: breakables
   * Extra - tile_types: ('WEIGHT',)
   > Door to Wave Beam Tutorial
-      Gravity Suit or After Ferenia - Wave Emmi Lower Water
+      Any of the following:
+          Gravity Suit or After Ferenia - Wave Emmi Lower Water
+          Lay Cross Comb and Perform WBJ
```
Comment (XenoWars) [inline randovania/games/dread/json_data/Ferenia.txt:2855]: i believe you can change the Lay Cross Bomb to Simple IBJ. Should be able to IBJ out of the water with normal bombs too

File: randovania/games/dread/json_data/Ferenia.txt [hunk 10]
```diff
@@ -2942,23 +2949,17 @@ Extra - asset_id: collision_camera_034
...
-      All of the following:
-          Before Ferenia - Central Unit
-          Gravity Suit or After Ferenia - Wave Emmi Lower Water
-          Space Jump or Infinite Bomb Jump (Beginner)
+  > Tunnel to Central Unit Access (Top)
+      Any of the following:
+          All of the following:
+              Gravity Suit and Before Ferenia - Wave Emmi Lower Water
+              Space Jump or Walljump (Intermediate) or Simple IBJ
+          Speed Booster and Speedbooster Conservation (Intermediate) and Disabled Door Lock Randomizer
```
Comment (XenoWars) [inline randovania/games/dread/json_data/Ferenia.txt:2957]: Might be able to do this as a beginner wall jump with spin boost

File: randovania/games/dread/json_data/Ferenia.txt [hunk 11]
```diff
@@ -2942,23 +2949,17 @@ Extra - asset_id: collision_camera_034
...
       After Ferenia - Central Unit and Can Slide
 
 > Lower-Left Alcove; Heals? False
   * Layers: default
-  > Start Point
-      All of the following:
-          Morph Ball and After Ferenia - Central Unit
-          Any of the following:
-              Simple IBJ or Use Spin Boost
-              Slide and Slide Jump (Intermediate) and Walljump (Intermediate)
```
Comment (XenoWars) [inline randovania/games/dread/json_data/Ferenia.txt:2961]: why did you remove this connection?

File: randovania/games/dread/json_data/Ferenia.txt [hunk 12]
```diff
@@ -2997,6 +2998,14 @@ Extra - asset_id: collision_camera_034
...
       Morph Ball
 
+> Tunnel to Central Unit Access (Middle); Heals? False
+  * Layers: default
+  * Morph Ball Tunnel to Central Unit Access/Tunnel to Purple EMMI Arena (Middle)
+  > Tile Group (POWERBOMB)
+      Morph Ball
+  > Upper-right Alcove
+      Trivial
+
```
Comment (XenoWars) [inline randovania/games/dread/json_data/Ferenia.txt:3008]: remove the connection to Upper-right.

File: randovania/games/dread/json_data/Ferenia.txt [hunk 13]
```diff
@@ -2856,6 +2861,8 @@ Extra - asset_id: collision_camera_034
   * Extra - actor_def: actordef:actors/logic/breakabletilegroup/charclasses/breakabletilegroup.bmsad
   * Extra - actor_layer: breakables
   * Extra - tile_types: ('POWERBOMB',)
+  > Tunnel to Central Unit Access (Middle)
+      Trivial
```
Comment (XenoWars) [inline randovania/games/dread/json_data/Ferenia.txt:2865]: add a trivial connection to Upper-Right Alcove

File: randovania/games/dread/json_data/Ferenia.txt [hunk 14]
```diff
@@ -3058,6 +3067,8 @@ Extra - asset_id: collision_camera_038_A
   * Extra - start_point_actor_def: actordef:actors/logic/startpoint/charclasses/startpoint.bmsad
   > Tunnel to Purple EMMI Arena (Bottom)
       After Ferenia - Wave Emmi Lower Water and Lay Bomb
```
Comment (XenoWars) [inline randovania/games/dread/json_data/Ferenia.txt:3069]: this doesnt need the lay bomb template as thats covered by the morph ball launcher dock weakness

**Review by XenoWars — CHANGES_REQUESTED** [review]

**Comment by XenoWars** [comment]
Ok, review up

**Comment by Hugoshido** [comment]
Add Wavebeam to speedboost method to the central unit access

File: randovania/games/dread/json_data/Ferenia.txt (hunk 13 above)
Comment (Hugoshido) [reply randovania/games/dread/json_data/Ferenia.txt:2860]: Check

File: randovania/games/dread/json_data/Ferenia.txt (hunk 12 above)
Comment (Hugoshido) [reply randovania/games/dread/json_data/Ferenia.txt:3003]: Check

File: randovania/games/dread/json_data/Ferenia.txt (hunk 14 above)
Comment (Hugoshido) [reply randovania/games/dread/json_data/Ferenia.txt:3064]: Check

File: randovania/games/dread/json_data/Ferenia.txt (hunk 11 above)
Comment (Hugoshido) [reply randovania/games/dread/json_data/Ferenia.txt:2955]: Added it back in. Don´t know y but I saw the Flashgates in reverse

File: randovania/games/dread/json_data/Ferenia.txt (hunk 8 above)
Comment (Hugoshido) [reply randovania/games/dread/json_data/Ferenia.txt:2667]: Check

File: randovania/games/dread/json_data/Ferenia.txt (hunk 6 above)
Comment (Hugoshido) [reply randovania/games/dread/json_data/Ferenia.txt:2293]: Check

File: randovania/games/dread/json_data/Ferenia.txt (hunk 5 above)
Comment (Hugoshido) [reply randovania/games/dread/json_data/Ferenia.txt:2012]: Check

File: randovania/games/dread/json_data/Ferenia.txt (hunk 7 above)
Comment (Hugoshido) [reply randovania/games/dread/json_data/Ferenia.txt:2573]: Yes you can walljump from a ledge grab. So I put walljump (beginner) in.

File: randovania/games/dread/json_data/Ferenia.txt (hunk 1 above)
Comment (Hugoshido) [reply randovania/games/dread/json_data/Ferenia.txt:152]: Spin boost is enough.. But it´s super tight.

File: randovania/games/dread/json_data/Ferenia.txt (hunk 2 above)
Comment (Hugoshido) [reply randovania/games/dread/json_data/Ferenia.txt:189]: With IBJ you barely come out of the water and would need SM to climb out. With Cross bombs you can just jump out. I would say IBJ is not enough

File: randovania/games/dread/json_data/Ferenia.txt (hunk 3 above)
Comment (Hugoshido) [reply randovania/games/dread/json_data/Ferenia.txt:193]: With IBJ you barely come out of the water and would need SM to climb out. With Cross bombs you can just jump out. I would say IBJ is not enough

File: randovania/games/dread/json_data/Ferenia.txt (hunk 9 above)
Comment (Hugoshido) [reply randovania/games/dread/json_data/Ferenia.txt:2850]: No it´s not possible with normal bombs.

File: randovania/games/dread/json_data/Ferenia.txt (hunk 10 above)
Comment (Hugoshido) [reply randovania/games/dread/json_data/Ferenia.txt]: Yeah it's possible.. I added it

File: randovania/games/dread/json_data/Ferenia.txt (hunk 2 above)
Comment (XenoWars) [reply randovania/games/dread/json_data/Ferenia.txt]: So do it from the platform?

File: randovania/games/dread/json_data/Ferenia.txt [hunk 15]
```diff
@@ -188,9 +186,16 @@ Extra - asset_id: collision_camera_003
...
+          Flash Shift or Gravity Suit or Spider Magnet or Use Spin Boost
+          Grapple Beam and Grapple Movement (Beginner)
+          Lay Cross Comb and Perform WBJ
+  > Door to EMMI Zone Exit Middle (Power)
+      Trivial
   > Door to EMMI Zone Exit Middle (Charge)
-      Grapple Beam or Gravity Suit or Use Spin Boost
+      Any of the following:
+          Flash Shift or Gravity Suit or Spider Magnet or Lay Cross Comb or Use Spin Boost
+          Grapple Beam and Grapple Movement (Beginner)
```
Comment (XenoWars) [inline randovania/games/dread/json_data/Ferenia.txt:198]: you forgot to do the "Perform WBJ" here, but im completely certain you can just IBJ normally from the center platform. I'm also sure a WBJ isnt even needed

File: randovania/games/dread/json_data/Ferenia.txt [hunk 16]
```diff
@@ -2312,7 +2304,9 @@ Extra - asset_id: collision_camera_028
   > Door to Save Station North
       After Ferenia - db_reg_b2_005
   > Event - Wave Blob (db_reg_b2_005)
-      Wave Beam
+      Any of the following:
+          Wave Beam
+          Pseudo-Wave Beam (Beginner) and Shoot Beam
```
Comment (XenoWars) [inline randovania/games/dread/json_data/Ferenia.txt:2309]: you didn't include the Slide here

File: randovania/games/dread/json_data/Ferenia.txt [hunk 17]
```diff
@@ -2936,12 +2938,13 @@ Extra - asset_id: collision_camera_034
...
-          Space Jump or Simple IBJ
-  > Tunnel to Central Unit Access
-      All of the following:
-          Before Ferenia - Central Unit
-          Gravity Suit or After Ferenia - Wave Emmi Lower Water
-          Space Jump or Infinite Bomb Jump (Beginner)
+  > Tunnel to Central Unit Access (Top)
+      Any of the following:
+          All of the following:
+              Gravity Suit and Before Ferenia - Wave Emmi Lower Water
```
Comment (XenoWars) [inline randovania/games/dread/json_data/Ferenia.txt:2944]: this should say "or After" not "and Before"

**Review by XenoWars — CHANGES_REQUESTED** [review]

File: randovania/games/dread/json_data/Ferenia.txt (hunk 16 above)
Comment (Hugoshido) [reply randovania/games/dread/json_data/Ferenia.txt:2309]: For me in the Data Visualizer it shows slide.. lol

File: randovania/games/dread/json_data/Ferenia.txt [hunk 18]
```diff
@@ -1423,6 +1434,12 @@ Extra - asset_id: collision_camera_017
   > Pickup (Power Bomb Tank)
       Morph Ball
 
+> Event - Twin Robot Fight; Heals? False
+  * Layers: default
+  * Event  Ferenia - Twin Robot Fight
+  > Start Point 1
+      After  Ferenia - Twin Robot Fight
+
```
Comment (XenoWars) [inline randovania/games/dread/json_data/Ferenia.txt:1442]: this should be trivial and will probably error out if it calls itself

**Review by XenoWars — CHANGES_REQUESTED** [review]

File: randovania/games/dread/json_data/Ferenia.txt [hunk 19]
```diff
@@ -3051,7 +3067,9 @@ Extra - asset_id: collision_camera_038_A
   * Extra - start_point_actor_name: SP_Checkpoint_EmmySancPhase2
   * Extra - start_point_actor_def: actordef:actors/logic/startpoint/charclasses/startpoint.bmsad
   > Tunnel to Purple EMMI Arena (Bottom)
-      After Ferenia - Wave Emmi Lower Water and Lay Bomb
+      After Ferenia - Wave Emmi Lower Water
+  > Tunnel to Purple EMMI Arena (Middle)
+      After Ferenia - Wave Emmi Lower Water
```
Comment (XenoWars) [inline randovania/games/dread/json_data/Ferenia.txt:3072]: This tunnel and the lower Purple EMMI Arena area are all that's left. This tunnel will need to require the EMMI being defeated.

**Review by XenoWars — CHANGES_REQUESTED** [review]

**Comment by XenoWars** [comment]
Just need to add the consolidated Wave EMMI event, probably called "Event - Purple EMMI"

File: randovania/games/dread/json_data/Ferenia.txt [hunk 20]
```diff
@@ -2825,165 +2782,12 @@ Extra - asset_id: collision_camera_034
...
-  > Tile Group (BOMB) 2
-      After Ferenia - db_reg_b2_010 and Can Slide
-  > Event - Wide Beam Block 2
-      Push Wide Beam Block
-  > Upper-right Alcove
-      Morph Ball and After Ferenia - Central Unit
-  > Pickup (Wave Beam)
-      After Ferenia - Central Unit
+              Gravity Suit or Lay Cross Comb
+              Spin Boost and Water Bomb Jump (Intermediate) and Lay Bomb
```
Comment (XenoWars) [inline randovania/games/dread/json_data/Ferenia.txt:2790]: Use Spin Boost. Can also get rid of the All of the Following

**Review by XenoWars — CHANGES_REQUESTED** [review]
It still shows the Start Points as deleted, which is probably why the failures are happening.

File: randovania/games/dread/json_data/Ferenia.txt [hunk 21]
```diff
@@ -2566,7 +2567,9 @@ Extra - asset_id: collision_camera_031
   > Door to EMMI Zone Exit East Access
       Can Slide
   > Tunnel to Purple EMMI Arena
-      Simple IBJ or Use Spin Boost
+      Any of the following:
+          Flash Shift or Walljump (Beginner) or Simple IBJ or Use Spin Boost
+          Slide and Slide Jump (Beginner)
```
Comment (XenoWars) [inline randovania/games/dread/json_data/Ferenia.txt:2572]: Is the walljump from the ledge grab? do you think it's hard enough to be in a trick?

Comment (Hugoshido) [reply randovania/games/dread/json_data/Ferenia.txt:2572]: It's not hard..But I think that most of the people won't think of a ledge grab to get up there. I would leave it with the trick just to set a difference between first time players and people who already played more seeds.

File: randovania/games/dread/json_data/Ferenia.txt [hunk 22]
```diff
@@ -1652,9 +1661,7 @@ Extra - asset_id: collision_camera_019
   > Event - Storm Missile Gate
       Activate Storm Missile Locks
   > Bomb Ledge
-      All of the following:
-          Screw Attack
-          Walljump (Beginner) or Use Spin Boost
+      Screw Attack
```
Comment (XenoWars) [inline randovania/games/dread/json_data/Ferenia.txt:1664]: In that case, this one should also account for that cause you have to use a ledge grab iirc

Comment (Hugoshido) [reply randovania/games/dread/json_data/Ferenia.txt:1664]: No you don't need a ledge grab there. You can just jump up

**Review by XenoWars — APPROVED** [review]
